services/venice-controller/src/main/java/com/linkedin/venice/pushmonitor/PushStatusCollector.java

```diff
@@ -0,0 +1,194 @@
+import com.linkedin.venice.meta.Version;
+import com.linkedin.venice.pushstatushelper.PushStatusStoreReader;
+import com.linkedin.venice.utils.Pair;
+import com.linkedin.venice.utils.concurrent.VeniceConcurrentHashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+
+public class PushStatusCollector {
+  private static final Logger LOGGER = LogManager.getLogger(PushStatusCollector.class);
+
+  private final Consumer<String> pushCompletedHandler;
+  private final BiConsumer<String, String> pushErrorHandler;
+  private final Map<String, TopicPushStatus> daVinciStoreTopicToPartitionCountMap = new VeniceConcurrentHashMap<>();
+
+  private final PushStatusStoreReader pushStatusStoreReader;
+
+  private final ReadWriteStoreRepository storeRepository;
+
+  private final int daVinciPushStatusScanPeriodInSeconds;
+
+  private final ScheduledExecutorService offlinePushCheckScheduler = Executors.newScheduledThreadPool(1);
+
+  public PushStatusCollector(
+      ReadWriteStoreRepository storeRepository,
+      PushStatusStoreReader pushStatusStoreReader,
+      Consumer<String> pushCompletedHandler,
+      BiConsumer<String, String> pushErrorHandler,
+      int daVinciPushStatusScanPeriodInSeconds) {
+    this.storeRepository = storeRepository;
+    this.pushStatusStoreReader = pushStatusStoreReader;
+    this.pushCompletedHandler = pushCompletedHandler;
+    this.pushErrorHandler = pushErrorHandler;
+    this.daVinciPushStatusScanPeriodInSeconds = daVinciPushStatusScanPeriodInSeconds;
+  }
+
+  public void start() {
+    offlinePushCheckScheduler
+        .scheduleAtFixedRate(this::scanDaVinciPushStatus, 0, daVinciPushStatusScanPeriodInSeconds, TimeUnit.SECONDS);
+  }
+
+  public void subscribeTopic(String topicName, int partitionCount) {
+    String storeName = Version.parseStoreFromKafkaTopicName(topicName);
+    Store store = storeRepository.getStore(storeName);
+    if (store == null) {
+      LOGGER.warn("Store {} not found in store repository, will not monitor Da Vinci push status", storeName);
+    } else {
+      if (store.isDaVinciPushStatusStoreEnabled() && Version.parseVersionFromKafkaTopicName(topicName) > 1) {
+        LOGGER.info("Will monitor Da Vinci push status for topic {}", topicName);
+        daVinciStoreTopicToPartitionCountMap.put(topicName, new TopicPushStatus(partitionCount));
+      }
+    }
+  }
+
+  public void unsubscribeTopic(String topicName) {
+    daVinciStoreTopicToPartitionCountMap.remove(topicName);
+  }
+
+  public void handleServerPushStatusUpdate(String topicName, ExecutionStatus executionStatus, String detailsString) {
+
+    TopicPushStatus topicPushStatus = daVinciStoreTopicToPartitionCountMap.get(topicName);
```

Comment:
If the DaVinci push status hasn't been collected, it will be `null`, and this class will act on the server push status right away, which may be wrong.

Can we have some extra logic there?
For example, if the DaVinci push status collection hasn't been done at all for a given topic, we should mark the DaVinci status as `unknown` or `progress`, and after collecting the DaVinci status at least once, we can update the aggregated status.

Reply:
I think for this line, the map (I need to rename the map) is populated when subscribeTopic is called. It will create an object. 
Also TopicPushStatus's getDaVinciStaus is getOrDefault(topic, UNKNOWN) as you are suggesting here.